File: binary_search/binary_search.cpp
#include <iostream>
using namespace std;
int binarySearch(int arr[],int size,int key){
    int start = 0;
    int end = size-1;

    int mid = start + (end-start)/2;
    while(start <= end){
        if(arr[mid] == key){
            return mid;
        }
        // go to right part
        if(key > arr[mid]){
            start = mid + 1;
        }
        else{
            end = mid - 1;
        }
        mid = start + (end-start)/2;
    }
    return -1;
}
int main(){
    int even[6] = {2,4,6,8,12,18};
    int odd[5] = {3,8,11,14,16};

    int index = binarySearch(even,6,19);
    cout<<"Index of 12 is "<<index<<endl;
   return 0;
}

File: binarysearch.rb
def binary_search(arr, target)
    left = 0
    right = arr.length - 1
  
    while left <= right
      mid = left + (right - left) / 2
      if arr[mid] == target
        return mid  
      elsif arr[mid] < target
        left = mid + 1
      else
        right = mid - 1
      end
    end
  
    return -1  
  end
  
  arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
  target = 5
  result = binary_search(arr, target)
  
  if result != -1
    puts "Element #{target} found at index #{result}"
  else
    puts "Element #{target} not found in the array"
  end
  